src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -234,6 +235,26 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
                 // been applied to the result; make sure it's also successful.
                 ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
 
+                if (responseContent == null)
+                {
+                    Type createdODataResultType = actionExecutedContext.Result.GetType().GetGenericArguments().Count() > 0 ?
+                        typeof(CreatedODataResult<>).MakeGenericType(actionExecutedContext.Result.GetType().GetGenericArguments()[0]) : null;
+
+                    Type actionResultType = actionExecutedContext.Result.GetType();
+
+                    // Get the entity object from CreatedODataResult<T> via reflection.
+                    // Use the entity object to create an instance of ObjectResult.
+                    if (createdODataResultType != null &&
+                        (actionResultType == createdODataResultType || createdODataResultType.IsAssignableFrom(actionResultType)))
+                    {
+                        object entity = ((PropertyInfo)createdODataResultType.GetProperty("Entity")).GetValue(actionExecutedContext.Result);
+                        responseContent = new ObjectResult(entity)
```

Comment:
Down below there's a statement like:
```c#
responseContent.Value = queryResult
```

What's the impact of modifying `responseContent.Value` in cases where the `responseContent` is actually created here as opposed to coming from the user?

Reply:
Also, is it necessary to create a new instance of `ObjectResult`, looking at the code below, most of it only uses the value of `responseContent.Value`, which I assume is the same `entity` in this case. Maybe we can just extract the value and use it directly instead of creating an instance of `ObjectResult` where one was not provided.

Reply:
1) There is no negative impact since technically w update the `actionExecutedContext.Result` with the created object.
2) I am creating the `responseContent` object since it's `null` in line `238`. 

I think casting `actionExecutedContext.Result` to `ObjectResult` was a way to restrict the type of results we can apply query options on. This PR ensure we also apply query options to `CreatedODataResult` which is not a subclass of `ObjectResult`.

With no restrictions we can use the `actionExecutedContext.Result` directly. So I will keep it that way for now.